docs/components/plugins/AppConfigSchemaPropertiesTable.test.js

```diff
@@ -0,0 +1,171 @@
+    type: 'string',
+    meta: {
+      bareWorkflow: "Edit the 'Display Name' field in Xcode",
+    },
+  },
+  androidNavigationBar: {
+    description: 'Configuration for the bottom navigation bar on Android.',
+    type: 'object',
+    properties: {
+      visible: {
+        description: 'Determines how and when the navigation bar is shown.',
+        type: 'string',
+        properties: {
+          always: {
+            description: 'Test sub-sub-property',
+            type: 'boolean',
+          },
+        },
+        enum: ['leanback', 'immersive', 'sticky-immersive'],
+      },
+      backgroundColor: {
+        description: 'Specifies the background color of the navigation bar. ',
+        type: 'string',
+        pattern: '^#|(&#x23;)\\d{6}$',
+        meta: {
+          regexHuman: "6 character long hex color string, eg: `'#000000'`",
+        },
+      },
+    },
+  },
+  intentFilters: {
+    description: 'Configuration for setting an array of custom intent filters in Android manifest.',
+    example: [
+      {
+        autoVerify: true,
+        data: {
+          host: '*.expo.io',
+        },
+      },
+    ],
+    exampleString: '\n [{ \n "autoVerify": true, \n "data": {"host": "*.expo.io" \n } \n }]',
+    type: 'array',
+    uniqueItems: true,
+    items: {
+      type: 'object',
+      properties: {
+        autoVerify: {
+          description:
+            'You may also use an intent filter to set your app as the default handler for links',
+          type: 'boolean',
+        },
+        data: {
+          type: ['array', 'object'],
+          items: {
+            type: 'object',
+            properties: {
+              host: { description: 'the host, e.g. `myapp.io`', type: 'string' },
+            },
+            additionalProperties: false,
+          },
+          properties: {
+            host: { type: 'string' },
+          },
+          additionalProperties: false,
+        },
+      },
+      additionalProperties: false,
+      required: ['action'],
+    },
+  },
+};
+
+test('correct property and subproperty indent styling', () => {
+  const { getByTestId } = render(
+    <SluggerContext.Provider value={new GithubSlugger()}>
+      <AppConfigSchemaPropertiesTable schema={testSchema} />
+    </SluggerContext.Provider>
+  );
+
+  const level_0_style = `
+    marginLeft: 12px,
+    display: 'block',
+    listStyleType: 'circle',
+    width: 'fit-content',
+    overflowX: 'visible',
+  `;
+
+  const level_1_style = `
+    marginLeft: 44px,
+    display: 'inline',
+    listStyleType: 'default',
+    width: 'fit-content',
+    overflowX: 'visible',
+  `;
+
+  const level_2_style = `
+    marginLeft: 76px,
+    display: 'inline',
+    listStyleType: 'circle',
+    width: 'fit-content',
+    overflowX: 'visible',
+  `;
+
+  //level-0 property
+  //the #### is added to link this property name by markdown and included in the data-testid
+  expect(getByTestId('#### `name`')).toHaveStyle(level_0_style);
```

Comment:
this seems a little bit fragile to test styles like this - we'd need to come and update this test any time we change them. an alternative way to do this without inlining the values is to use snapshots. another way to do this would be to test that there is some attribute set that is more generic that won't change, or that the component type is one that corresponds to a level 0 element

Reply:
okay sounds good! do you know where I could get more information about snapshots?

And currently, the properties in `level_0_style` are all ones that I set within `AppConfigSchemaPropertiesTable.js` so I am not sure there is a more generic attribute set -> so snapshots might be the best solution right now

Reply:
this is a good doc for that: https://jestjs.io/docs/en/snapshot-testing

you can also search `toMatchSnapshot()` in our codebase to find other examples of usage